lib/core/application/event_tickets/modify_reward_bloc/modify_reward_bloc.dart

```diff
@@ -0,0 +1,201 @@
+import 'package:app/core/domain/event/entities/event_ticket_types.dart';
+import 'package:app/core/domain/event/entities/reward.dart';
+import 'package:app/core/domain/event/repository/event_reward_repository.dart';
+import 'package:app/core/domain/form/string_formz.dart';
+import 'package:app/graphql/backend/schema.graphql.dart';
+import 'package:app/injection/register_module.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:formz/formz.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+part 'modify_reward_bloc.freezed.dart';
+
+class ModifyRewardBloc extends Bloc<ModifyRewardEvent, ModifyRewardState> {
+  ModifyRewardBloc() : super(ModifyRewardState.initial()) {
+    on<_ModifyRewardEventOnTitleChanged>(_onTitleChanged);
+    on<_ModifyRewardEventOnIconChanged>(_onIconChanged);
+    on<_ModifyRewardEventOnLimitPerChanged>(_onLimitPerChanged);
+    on<_ModifyRewardEventOnToggleSelectAll>(
+      _onToggleSelectAll,
+    );
+    on<_ModifyRewardEventOnSelectTicketTier>(
+      _onSelectEventTicketType,
+    );
+    on<_ModifyRewardEventOnValidate>(_onValidate);
+    on<_ModifyRewardEventOnSubmitted>(_onSubmitted);
+  }
+  final _eventRepository = getIt<EventRewardRepository>();
+
+  void _onTitleChanged(
+    _ModifyRewardEventOnTitleChanged event,
+    Emitter emit,
+  ) {
+    final newState = state.copyWith(
+      title: StringFormz.dirty(event.title),
+    );
+    emit(_validate(newState));
+  }
+
+  void _onIconChanged(
+    _ModifyRewardEventOnIconChanged event,
+    Emitter emit,
+  ) {
+    final newState = state.copyWith(
+      iconUrl: event.iconUrl,
+    );
+    emit(_validate(newState));
+  }
+
+  void _onLimitPerChanged(
+    _ModifyRewardEventOnLimitPerChanged event,
+    Emitter emit,
+  ) {
+    final newState = state.copyWith(
+      limitPer: event.limitPer,
+    );
+    emit(_validate(newState));
+  }
+
+  void _onToggleSelectAll(
+    _ModifyRewardEventOnToggleSelectAll event,
+    Emitter emit,
+  ) {
+    List<String> eventTicketTypesIds =
+        event.eventTicketTypes.map((obj) => obj.id ?? '').toList();
+    if (event.value == true) {
+      final newState = state.copyWith(
+        selectedEventTicketTypeIds: eventTicketTypesIds,
+      );
+      emit(_validate(newState));
+    } else {
+      final newState = state.copyWith(
+        selectedEventTicketTypeIds: [],
+      );
+      emit(_validate(newState));
+    }
+  }
+
+  void _onSelectEventTicketType(
+    _ModifyRewardEventOnSelectTicketTier event,
+    Emitter emit,
+  ) {
+    if (state.selectedEventTicketTypeIds.contains(event.eventTicketTypeId)) {
```

Comment:
This action contains two logic: select and deselect. I think we can separate into two functions which is more readable